### PR TITLE
improved search method (loading properties bug fixed, swagger ui

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
improved)

### CODE REVIEW

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationSearchRest.java [hunk 1]
```diff
@@ -134,30 +147,47 @@ private ResponseEntity<String> searchAnnotation(String wskey, String queryString
...
+              new String[] { WebAnnotationFields.PARAM_PROFILE, profile });
+        }
+      }
+  }
+
+    private SearchProfiles getProfile(String profile, HttpServletRequest request) throws HttpException {
+      List<String> allProfiles = GeneralUtils.splitStringIntoList(profile, WebAnnotationModelFields.COMMA);
+      validateProfiles(allProfiles);      
+      allProfiles.remove(SearchProfiles.FACETS.toString());
+      if(allProfiles.size()>0) {
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationSearchRest.java:176]: we should take care of profile=debug here ... the method should return a list of profiles ...

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationSearchRest.java [hunk 2]
```diff
@@ -134,30 +147,47 @@ private ResponseEntity<String> searchAnnotation(String wskey, String queryString
...
+        preferHeader = preferHeader.replaceAll("\\s+", "");
+    	if (preferHeader.equals(AnnotationHttpHeaders.VALUE_PREFER_CONTAINEDIRIS)) {
+    	  return SearchProfiles.MINIMAL;
+    	} else if (preferHeader.equals(AnnotationHttpHeaders.VALUE_PREFER_CONTAINEDDESCRIPTIONS)) {
+    	  return SearchProfiles.STANDARD;
+    	}
+      }
+
+      logger.trace("STANDARD Profile set by default");
+      return SearchProfiles.STANDARD;
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationSearchRest.java:191]: the default profile needs to be added also when the profile=debug is present in the request

File: annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java [hunk 3]
```diff
@@ -167,12 +170,38 @@ public void testSearchAnyAnnotationMinimalProfile() throws Exception {
...
+  @Test
+  public void searchWithFacets() throws Exception {
+    createAnnotationDataSet();
+
+    String[] facets = {"motivation", "generator_uri"};
+    AnnotationPage annPg = searchAnnotations("*:*", null, facets, null, null, null, null, "minimal", null);
+    assertNotNull(annPg, "AnnotationPage must not be null");
+    assertNull(annPg.getFacets());
+
+    annPg = searchAnnotations("*:*", null, facets, null, null, null, null, "minimal, facets", null);
```
[the remark this answers is not in the document]
Comment (gsergiu) [reply annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java:199]: there should be another test case for profile=debug in which case the default standard profile should be used for serialization. The presence of the motivation field can be used to verify if the standard serialization was used

File: api2-common/src/main/resources/config/annotation.user.properties.template [hunk 4]
```diff
@@ -1,6 +1,11 @@
 ### annotation database ###
 # mongoDB settings
 mongodb.annotation.connectionUrl=mongodb://127.0.0.1:27017/annotation_test
+mongodb.annotation.host=localhost
```
Comment (gsergiu) [inline api2-common/src/main/resources/config/annotation.user.properties.template:4]: the code in the develop branch was updated to use only the connection URL, the rest of the properties must be removed. Please merge the develop branch and revert the changes in this file

**Review by gsergiu — CHANGES_REQUESTED** [review]

File: annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SrdjanStevanetic) [reply annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java:199]: Added tests for profile=debug and profile=minimal,factes,debug. The search with different profiles is done in one test, no need for an overhead of creating 21 annotations for each test, just because the search profile changed.

File: annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java [hunk 5]
```diff
@@ -432,9 +475,9 @@ protected Annotation searchLastCreated(String query) throws Exception {
     return anno;
   }
 
-  protected AnnotationPage search(String bodyValue, SearchProfiles profile, String limit)
+  protected AnnotationPage search(String bodyValue, String profile, String limit)
       throws Exception {
-    AnnotationPage annPg = searchAnnotations(bodyValue, null, WebAnnotationFields.CREATED, "desc",
+    AnnotationPage annPg = searchAnnotations(bodyValue, null, null, WebAnnotationFields.CREATED, "desc",
```
Comment (gsergiu) [inline annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java:480]: shouldn't be the profile passed over to the search method at this place?

Comment (SrdjanStevanetic) [reply annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java:480]: It is passed in the before last parameter.

Comment (gsergiu) [reply annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java:480]: it is passed null instead of profile ...
